src/webgpu/shader/execution/statement/increment_decrement.spec.ts

```diff
@@ -0,0 +1,301 @@
+      new Uint32Array([0]),
+      `
+    var a: u32 = ${kValue.u32.max};
+    a++;
+    push_output(a);
+`
+    );
+  });
+
+g.test('scalar_i32_decrement')
+  .desc('Tests decrement of scalar i32 values')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'i32',
+      new Int32Array([-11, 9, kValue.i32.negative.min, kValue.i32.positive.max - 1, -1]),
+      `
+    var a: i32 = -10;
+    var b: i32 = 10;
+    var c: i32 = ${kValue.i32.negative.min + 1};
+    var d: i32 = ${kValue.i32.positive.max};
+    var e: i32 = 0;
+
+    a--;
+    b--;
+    c--;
+    d--;
+    e--;
+
+    push_output(a);
+    push_output(b);
+    push_output(c);
+    push_output(d);
+    push_output(e);
+`
+    );
+  });
+
+g.test('scalar_i32_decrement_underflow')
+  .desc('Tests decrement of scalar i32 values which underflow')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'i32',
+      new Int32Array([kValue.i32.positive.max]),
+      `
+    var a: i32 = ${kValue.i32.negative.min};
+    a--;
+    push_output(a);
+`
+    );
+  });
+
+g.test('scalar_u32_decrement')
+  .desc('Tests decrement of scalar u32 values')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'u32',
+      new Uint32Array([0, 9, kValue.u32.max - 1]),
+      `
+    var a: u32 = 1;
+    var b: u32 = 10;
+    var c: u32 = ${kValue.u32.max};
+
+    a--;
+    b--;
+    c--;
+
+    push_output(a);
+    push_output(b);
+    push_output(c);
+`
+    );
+  });
+
+g.test('scalar_u32_decrement_underflow')
+  .desc('Tests decrement of scalar u32 values which underflow')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'u32',
+      new Uint32Array([kValue.u32.max]),
+      `
+    var a: u32 = 0;
+    a--;
+    push_output(a);
+`
+    );
+  });
+
+g.test('vector_element_increment')
+  .desc('Tests increment of vector values')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'i32',
+      new Int32Array([-9, 11, kValue.i32.negative.min + 1, kValue.i32.positive.max]),
+      `
+    var a: vec4<i32> = vec4(-10, 10, ${kValue.i32.negative.min}, ${kValue.i32.positive.max - 1});
+
+    a.x++;
+    a.g++;
+    a.z++;
+    a.a++;
+
+    push_output(a.x);
+    push_output(a.y);
+    push_output(a.z);
+    push_output(a.w);
+`
+    );
+  });
+
+g.test('vector_element_decrement')
+  .desc('Tests decrement of vector values')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'i32',
+      new Int32Array([-11, 9, kValue.i32.negative.min, kValue.i32.positive.max - 1]),
+      `
+    var a: vec4<i32> = vec4(-10, 10, ${kValue.i32.negative.min + 1}, ${kValue.i32.positive.max});
+
+    a.x--;
+    a.g--;
+    a.z--;
+    a.a--;
+
+    push_output(a.x);
+    push_output(a.y);
+    push_output(a.z);
+    push_output(a.w);
+`
+    );
+  });
+
+g.test('frexp_exp_increment')
+  .desc('Tests increment can be used on a frexp field')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'i32',
+      new Int32Array([2]),
+      `
+    var a = frexp(1.23);
+    a.exp++;
+    push_output(a.exp);
+`
+    );
+  });
```

Comment:
Note to myself: There are not matrix tests, because matrices cannot contain integer types.